src/test/pybind/test_rbd.py

```diff
@@ -324,6 +324,17 @@ def test_list():
         image_id = image.id()
     eq([{'id': image_id, 'name': image_name}], list(RBD().list2(ioctx)))
 
+@with_setup(create_image)
+def test_remove_with_progress():
+    received_callback = False
+    def progress_cb(current, total):
+        nonlocal received_callback
+        received_callback = True
+        return 0
```

Comment:
It would be nice to have a test that returns a negative error code from the callback.